### PR TITLE
Allow evaluation of last datapoint only for simple patterns

### LINKED ISSUE #37: Revert evaluating full requestContextInterval
See https://github.com/moira-alert/worker/commit/e57ea9562ae96fe2c410f91a36bc096c8ce44a33 and especially https://github.com/moira-alert/worker/commit/e57ea9562ae96fe2c410f91a36bc096c8ce44a33#diff-ad6e777f8d2acc24d503350da3faefe5R134.

After some investigation I came to believe that adding an extra bucket at the end of timeseries is a mistake.

Consider two metrics with names `cpu1`, `cpu2` and function `movingAverage(cpu*, '3min', 'min')`. Let's say that `cpu1` sends points several seconds earlier than `cpu2`. At some point in time the timeseries could look like this:

`cpu1: [1, 2, 2, 2]`
`cpu2: [1, 2, 2, null]` - last point hasn't arrived yet

Original Graphite logic is to always drop the last point, so our `movingAverage` will compute 1 from `[1, 2, 2]` for both metrics. Current Moira logic is to always consider the last point, even if it hasn't arrived yet, so our `movingAverage` will compute 2 from `[2, 2, 2]` and `[2, 2, null]`.

After some time actual value for `cpu2` arrives, and suddenly...

`cpu1: [1, 2, 2, 2]`
`cpu2: [1, 2, 2, 1]`

...Moira logic generated a false alarm for `cpu2` :(

A possible workaround is to include last bucket only if it contains a datapoint. But this is the worst option, because it shifts some metrics forward and corrupts all combining functions like `sumSeries`.

I think we should either:

1. Revert the whole commit in question.
2. Remove `+ retention` from line 134.

@gmlexx it's up to you :)

### PR DESCRIPTION
**Pull request by beevee**
Fixes #37

### CODE REVIEW

File: moira/graphite/datalib.py [hunk 1]
```diff
@@ -134,7 +134,11 @@ def fetchData(requestContext, pathExpr):
                 bucket = (int)((timestamp - startTime) / retention)
                 points[bucket] = extract(value)
 
-            values = [points.get((int)((timestamp - startTime) / retention)) for timestamp in xrange(startTime, endTime + retention, retention)]
+            values = [points.get((int)((timestamp - startTime) / retention)) for timestamp in xrange(startTime, endTime, retention)]
```
Comment (gmlexx) [inline moira/graphite/datalib.py]: You cut the last point here

Comment (beevee) [inline moira/graphite/datalib.py]: Yes I do! But I re-add it back later.

**Comment by beevee** [comment]
This is proof-of-concept now, wait for further fixes.

**Comment by gmlexx** [comment]
Why you didn't start from functional tests?

**Comment by gmlexx** [comment]
I thought, that we need to investigate internal functions like movingAverage behavior is problem cases firstly with some functional tests

**Pull request closed without merging**